source/brailleDisplayDrivers/albatross.py

```diff
@@ -0,0 +1,822 @@
+
+	def start(self):
+		if not self.is_running:
+			self._timer = Timer(self.interval, self._run)
+			self._timer.start()
+			self.is_running = True
+
+	def stop(self):
+		self._timer.cancel()
+		self.is_running = False
+
+
+class BrailleDisplayDriver(braille.BrailleDisplayDriver):
+	name = "albatross"
+	# Translators: Names of braille displays.
+	description = _("Caiku Albatross 46/80")
+	isThreadSafe = True
+
+	@classmethod
+	def getManualPorts(cls):
+		return braille.getSerialPorts()
+
+	def __init__(self, port="Auto"):
+		super().__init__()
+		# Number of cells is received when initializing connection.
+		self.numCells = 0
+		# Keep old display data, only changed content is sent.
+		self._oldCells: List[int] = []
+		# After reconnection and when user exits from device menu, display may not
+		# update automatically. Keep current cell content for this.
+		self._currentCells: List[int] = []
+		# Set to True if settings byte is INIT_START_BYTE
+		self._invalidSettingsByte = False
+		# Set to True when filtering redundant init packets when init byte
+		# received but not yet settings byte.
+		self._initByteReceived = False
+		# Set to True if settings byte has not been dequeued yet in
+		# _handleInitPackets.
+		self._waitingSettingsByte = False
+		# Set to True if ctrl key combination is partially dequeued.
+		self._waitingCtrlPacket = False
+		# Port object
+		self._dev = None
+		# Try to connect or reconnect
+		self._tryToConnect = False
+		self._readQueue = deque()
+		self._writeQueue = deque(maxlen=WRITE_QUEUE_LENGTH)
+		self._exitEvent = Event()
+		# Thread for read
+		self._handleRead = None
+		self._writeLock = Lock()
+		# Timer to keep connection (see KC_INTERVAL).
+		self._kc = None
+		# When previous write was done (see KC_INTERVAL).
+		self._writeTime = 0.0
+		# Search ports where display can be connected.
+		for portType, portId, port, portInfo in self._getTryPorts(port):
+			# For reconnection
+			self._currentPort = port
+			self._tryToConnect = True
+			self._readHandling()
+			if self.numCells:
+				# Prepare _oldCells to store last displayed content.
+				while len(self._oldCells) < self.numCells:
+					self._oldCells.append(0)
+				self._kc = RepeatedTimer(
+					KC_INTERVAL, self._keepConnected)
+				self._handleRead = ReadThread(
+					self._readHandling, self._disableConnection, self._exitEvent,
+					self._dev, name="albatross_read", daemon=True)
+				self._handleRead.start()
+				log.info(
+					f"Connected to Caiku Albatross {self.numCells} on {portType} port {port} "
+					f"at {BAUD_RATE} bps.")
+				break
+			# This device initialization failed.
+			if self._dev:
+				if self._dev.is_open:
+					self._dev.close()
+				self._dev = None
+			if self._invalidSettingsByte:
+				log.info(
+					"After checking internal settings, switch display off and on, "
+					"and if needed restart NVDA")
+			log.info(
+				f"Connection to Caiku Albatross display on {portType} port {port} "
+				f"at {BAUD_RATE} bps failed.")

```

Comment:
Please fix up all these hanging parentheses to a new line

```suggestion
			log.info(
				f"Connection to Caiku Albatross display on {portType} port {port} "
				f"at {BAUD_RATE} bps failed."
			)
```